microcosm_pubsub/matchers.py

```diff
@@ -0,0 +1,95 @@
+from dataclasses import dataclass
+from json import loads
+from typing import List, Optional, Mapping
+
+from hamcrest import not_none
+from hamcrest.library.collection.issequence_containinginorder import IsSequenceContainingInOrder
+from hamcrest.library.object.hasproperty import IsObjectWithProperty
+from hamcrest.core.helpers.wrap_matcher import wrap_matcher
+
+
+@dataclass
+class PublishedMessage:
```

Comment:
Having an object to encapsulate the `mock_call` allows us to use existing object matchers with the internal structure (and abstract out the JSON decoding).